remotespark/livyclientlib/livysession.py

```diff
@@ -68,13 +68,21 @@ def create_sql_context(self):
 
         self.logger.debug("Starting '{}' sql and hive session.".format(self.language))
 
-        self.wait_for_status(Constants.idle_session_status, self._create_sql_context_timeout_seconds)
-        self.execute(self._get_sql_context_creation_command())
-        self.logger.debug("Started '{}' sql session.".format(self.language))
-
-        self.wait_for_status(Constants.idle_session_status, self._create_sql_context_timeout_seconds)
-        self.execute(self._get_hive_context_creation_command())
-        self.logger.debug("Started '{}' hive session.".format(self.language))
+        try:
+            self.wait_for_status(Constants.idle_session_status, self._create_sql_context_timeout_seconds)
+            self.execute(self._get_sql_context_creation_command())
+            self.logger.debug("Started '{}' sql session.".format(self.language))
+        except LivyClientTimeoutError:
+            raise LivyClientTimeoutError("Failed to create the SQL context in time. Timed out after {} seconds."
+                                         .format(self._create_sql_context_timeout_seconds))
+
+        try:
+            self.wait_for_status(Constants.idle_session_status, self._create_sql_context_timeout_seconds)
```

Comment:
Copy-paste here. This and the above try-except block should probably be refactored into a separate method